script.js

```diff
@@ -1,4 +1,244 @@
-var main = function (input) {
-  var myOutputValue = 'hello world';
+// ################ MAKE DECK ################
+function makeDeck() {
+  // Initialise an empty deck array
+  var cardDeck = [];
+  // Initialise an array of the 4 suits in our deck. We will loop over this array.
+  var suits = ["spades", "hearts", "clubs", "diamonds"];
+
+  // Loop over the suits array
+  for (suitIndex = 0; suitIndex < suits.length; suitIndex++) {
+    // Store the current suit in a variable
+    var currentSuit = suits[suitIndex];
+
+    // Loop from 1 to 13 to create all cards for a given suit
+    // Notice rankCounter starts at 1 and not 0, and ends at 13 and not 12.
+    for (rankCounter = 1; rankCounter <= 13; rankCounter++) {
+      // By default, the card name is the same as rankCounter
+      var cardName = rankCounter;
+      var cardValue = rankCounter;
+
+      // If rank is 1, 11, 12, or 13, set cardName to the ace or face card's name and set cardValue accordingly
+      if (cardName == 1) {
+        cardName = "ace";
+      } else if (cardName == 11) {
+        cardName = "jack";
+        cardValue = 10;
+      } else if (cardName == 12) {
+        cardName = "queen";
+        cardValue = 10;
+      } else if (cardName == 13) {
+        cardName = "king";
+        cardValue = 10;
+      }
+
+      // Create a new card with the current name, suit, and rank
+      var card = {
+        name: cardName,
+        suit: currentSuit,
+        rank: rankCounter,
+        value: cardValue,
+      };
+
+      // Add the new card to the deck
+      cardDeck.push(card);
+    }
+  }
+
+  // Return the completed card deck
+  return cardDeck;
+}
+
+// ================ randomiser ================
+function randomiser(max) {
+  return Math.floor(Math.random() * max);
+}
+
+// ################ SHUFFLE DECK ################
+function shuffleCards(cardDeck) {
+  // Loop over the card deck array once
+  for (index in cardDeck) {
```

Comment:
Really concise way of creating a for loop! Just one note that for..in should be avoided in cases where the sequence is important it does not guarantee order. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in#array_iteration_and_for...in